README.md

```diff
@@ -1,12 +1,149 @@
+```
+store(document: User & Partial<Document>): Promise<(User & Document) | undefined>
+store(documents: (User & Partial<Document>)[]): Promise<((User & Document) | undefined)[]>
+```
+### load
+Loads document(s) form the database by either specifying the id, ids or a query in the form specified in the type `Database/Selection/index.ts`. A locus can be returned if the query includes more documents than the specified limit or 1000 as default. The locus can be used to continue the listing from where the previous one stopped.
+
+```
+load(id: Identifier): Promise<(T & Document) | undefined>
+load(ids: Identifier[]): Promise<((Document & T) | undefined)[] & { locus?: string }>
+load(selection?: Selection): Promise<(Document & T)[] & { locus?: string }>
+```
+### remove
+The remove function will delete the document and return a boolean representing the success of the operation.
+```
+remove(id: Identifier): Promise<boolean>
+remove(id: Identifier[]): Promise<boolean[]>
+```
+### partition
+The partition function can be used to create partitions, theoretically infinitly many partitions can be created but don't go crazy with it.
+```
+partition(...partition: Identifier[]): S 
+```
+## Indecies
+There exists 3 types of keys in the Database, `id`, `changed` and `doc`; each adds functionality to the database.
+The definitions below are based on how they look in the `Archive`; the buffer uses a small variation of these keys where the type of the document as well as the slash following the type is omitted.
+
+### doc
+The doc-index contains the document the user wants to store/load/update.
+The structure of the doc-key makes it possible, easy and fast to list ceratin a partition and query it on created time with a date range.
+
+Definition:
+```
+key := type + "/doc/" + partitions.join("/") + "/" + document.created + "/" + document.id
+{
+	[key]: document
+}
+```
+Where type is the name of the document used in the layout when initiating the database, the partitions are all of the partitions added to the initiated database, the document is the value the user wants to store.
+
+
+
+### id
+The purpose of the id-index is to make it possible to fetch a document while only knowing the type and id of the document.
+Definition: 
+```
+{
+	[type + "/id/" + document.id]: key
+}
+```
+Where the type is the name of the document-type used in the layout when initiating the database and the document is the value the user wants to store and the key is defined above in `doc`.
+
+### changed 
+The changed-index is multi purposed and is used in the buffer to determine which documents to archive and which archived documents to remove. Its use in the archive is to be able query documents with a daterange representing the last time the document was changed.
+Definition: 
+```
+{
+	changed := truncate(document.changed, "minutes")
+	[type + "/changed/" + changed]: `key_1\n
+																	 key_2\n
+																	 ...
+																	 key_n`
+}
```

Comment:
Add a tab and then spaces if you want `key_` to align when shown in monospace, since tabs are not the same witdh everywhere.
```suggestion
{
	changed := truncate(document.changed, "minutes")
	[type + "/changed/" + changed]: `key_1\n
	                                 key_2\n
	                                 ...
	                                 key_n`
}
```